### PR TITLE
[Azure Search] Fixing some long-lived bugs in the data plane SDK

### PR DESCRIPTION
**Pull request by bitsforthought**
## Description

This PR fixes a few long-standing bugs in the Azure Search data plane SDK. You won't see it in the commits, but one of the bugs fixed is the [incompatibility with JSON.NET 10.0.1](https://github.com/Azure/azure-sdk-for-net/issues/3441). This is fixed just by virtue of rebuilding the SDK with the latest `ClientRuntime` on the correct branch (`psSdk6Json` instead of `vs17Dev`).

The other fixes are for these issues:
  - [Custom HttpClientHandler not called on Search requests](https://github.com/Azure/azure-sdk-for-net/issues/3451)
  - [FieldBuilder tries to read a property marked with JsonIgnore](https://github.com/Azure/azure-sdk-for-net/issues/3309)

There are no Swagger changes and no breaking changes in this PR. This PR bumps the SDK version number from 3.0.4 to 3.0.5.

See individual commits for more details.

FYI @Yahnoosh @mhko @updixit @shahabhijeet 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs [hunk 1]
```diff
@@ -53,17 +54,22 @@ public static IList<Field> BuildForType<T>(IContractResolver contractResolver)
             var fields = new List<Field>();
             foreach (JsonProperty prop in contract.Properties)
             {
+                IList<Attribute> attributes = prop.AttributeProvider.GetAttributes(true);
+                if (attributes.Any(attr => attr is JsonIgnoreAttribute))
+                {
+                    continue;
+                }
+
```
Comment (Yahnoosh) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs]: should we add a regression test for this?

**Review by Yahnoosh — APPROVED** [review]
:shipit:

**Comment by shahabhijeet** [comment]
@azuresdkci  retest this please

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs (hunk 1 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs]: @Yahnoosh There are already test changes for this. Look at the test model class.

Comment (Yahnoosh) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/FieldBuilder.cs]: thanks

---
In reply to: [166069859](https://github.com/Azure/azure-sdk-for-net/pull/4036#discussion_r166069859) [](ancestors = 166069859)

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**